index.js

```diff
@@ -54,43 +54,63 @@ function fastErrorHTTP(code, status) {
 }
 
 /**
- * An express.js middleware used for showing errors in your logs
+ * Creates an express.js middleware used for showing errors in your logs
  * and converting them to JSONP enabled response messages.
+ * 
+ * The options object allows defining a custom logger for the middleware.
  *
  * It will log any >=500 status codes using fastlog and update the
  * error message to "Internal Server Error" to obfuscate any stack or
  * native-level errors that you don't want to expose to users.
  *
  * Any status codes <500 are assumed to contain error messages crafted
  * for public consumption.
+ * @param {Object} options - Options for the middleware
+ * @param {Object} options.logger Custom logger for logging the errors
+ * @param {Function} options.logger.error
  */
-
-// NOTE: next is needed, even if not used, per https://expressjs.com/en/guide/using-middleware.html
-function showError(err, req, res, next) { // eslint-disable-line no-unused-vars
-  err.status = err.status || 500;
-
-  // Output unexpected errors to console but hide them from public eyes.
-  if (err.status >= 500) {
-    err.url = req.url;
-    err.method = req.method;
-    logger.error(err);
-    err.message = 'Internal Server Error';
-  }
-
-  const data = { message:err.message };
-
-  // For non-500 ErrorHTTP objects send over any additional keys.
-  // This error is one that we crafted ourselves deliberately for
-  // public consumption.
-  if (err instanceof ErrorHTTP && err.status < 500) {
-    for (const k in err) {
-      if (k === 'status') continue;
-      data[k] = err[k];
+function showErrorWithOptions(options = {}) {
+  const logger = options.logger || fastlogLogger;
```

Comment:
Can the default value for options param be `{ logger: fastlogLogger }`  instead? That will remove this line and also you won't have to pass the object for showError const below

Reply:
The default logger would not be used if the user passes any option object, even an empty one {}. Unfortunately there's no way to set defaults for single values within an object.

What happens is:
```
const middleware = showErrorWithOptions({}); // options.logger is undefined
// or
const middleware = showErrorWithOptions({ someOtherConfiguration: true }); // options.logger is undefined
```

To prevent this we would still need to have the line for `const logger = options.logger || fastlogLogger;` which defeats the purpose.

Reply:
makes sense. Thanks